File: nysol_python/mcmd/methods/convTex/db/smp/func_capitalize.py
# ================================================
# コマンドマニュアル自動作成用サンプルコードデータ
# ================================================
db={}
db['name']='capitalize'

############################### IDATA
db['idatas']=[]

data={}
data['name']='dat1.csv'
data['text']='''
id,str
1,abc
2,aBd
3,
4,#abc
'''
db['idatas'].append(data)

############################### SCRIPTS
db['scripts']=[]

script={}
script['title']='基本例'
script['comment']='''
$str$項目の先頭文字を大文字に変換する。
'''
script['sh_code']='''
mcal c='capitalize($s{str})' a=rsl i=dat1.csv o=rsl1.csv
'''
script['py_code']='''
nm.mcal(c='capitalize($s{str})', a='rsl', i="dat1.csv", o="rsl1.csv").run()
'''
script['odatas']='rsl1.csv'
db['scripts'].append(script)

File: olddoc/mcmd/en/examples/func_capitalize.rb
#!/usr/bin/env ruby
# coding: utf-8
require "./mkTex.rb"

File.open("dat1.csv","w"){|fpw| fpw.write(
<<'EOF'
id,str
1,abc
2,aBd
3,
4,#abc
EOF
)}

############## Example 1
title="Basic Example"
comment=<<'EOF'
Convert the first character of $str$ field to uppercase.
EOF
scp=<<'EOF'
more dat1.csv
mcal c='capitalize($s{str})' a=rsl i=dat1.csv o=rsl1.csv
more rsl1.csv
EOF
run(scp,title,comment)

